src/TodoApp.js

```diff
@@ -0,0 +1,137 @@
+import Component from "./core/Component.js";
+import TodoAppender from "./components/TodoAppender.js";
+import TodoList from "./components/TodoList.js";
+import TodoFilter from "./components/TodoFilter.js";
+export default class TodoApp extends Component {
+  setup() {
+    this.$state = localStorage.getItem("state")
+      ? JSON.parse(localStorage.getItem("state"))
+      : {
+          typeOfFilter: "all",
+          todos: [
+            {
+              seq: 1,
+              contents: "todo 1",
+              state: "doing",
+              edit: "",
+            },
+            {
+              seq: 2,
+              contents: "todo 2",
+              state: "doing",
+              edit: "",
+            },
+            {
+              seq: 3,
+              contents: "todo 3",
+              state: "completed",
+              edit: "",
+            },
+          ],
+        };
+  }
+  template() {
+    return `
+      <h1>TODOS</h1>
+      <input id="new-todo-title" class="new-todo" placeholder="할일을 추가해주세요" autofocus></input>
+      <main>
+        <input class="toggle-all" type="checkbox"/>
+          <ul id="todo-list" class="todo-list"></ul>
+          <div id="todo-filter" class="count-container"></div>
+      </main>
+    `;
+  }
+
+  mounted() {
+    const {
+      filteredList,
+      addTodo,
+      toggleTodo,
+      deleteTodo,
+      onEditingMode,
+      editTodo,
+      filterList,
+    } = this;
+    const $todoAppender = this.$target.querySelector("#new-todo-title");
+    const $todoList = this.$target.querySelector("#todo-list");
+    const $todoFilter = this.$target.querySelector("#todo-filter");
+    new TodoAppender($todoAppender, {
+      addTodo: addTodo.bind(this),
+    });
+
+    new TodoList($todoList, {
+      filteredList,
+      toggleTodo: toggleTodo.bind(this),
+      deleteTodo: deleteTodo.bind(this),
+      onEditingMode: onEditingMode.bind(this),
+      editTodo: editTodo.bind(this),
+    });
+
+    new TodoFilter($todoFilter, {
+      typeOfFilter: this.$state.typeOfFilter,
+      filteredList,
+      filterList: filterList.bind(this),
+    });
+  }
+
+  get filteredList() {
+    const { typeOfFilter, todos } = this.$state;
+    return todos.filter(({ state }) => {
+      return typeOfFilter === "all" || typeOfFilter === state;
+    });
+  }
+
+  addTodo(contents) {
+    const { todos } = this.$state;
+    const seq = Math.max(0, ...todos.map((todo) => todo.seq)) + 1;
```

Comment:
아마도 고유한 정수값을 ```id```로 부여하시기 위해 위와 같은 방법을 사용하신거 같은데 아래와 같이 ```Date.now()``` 메서드를 사용하시면 더 쉽게 고유값으로 아이디를 만드실 수 있을거 같습니다!
```suggestion
    const seq = Date.now()
```

Reply:
아 그리고 혹시 흔히 쓰는 ```id```라는 이름 대신 ```seq```라는 이름을 사용하신 이유가 궁금합니다!

Reply:
sequence의 줄임말로 id겸 '순서'를 나타냅니다. 그래서 저 Math.max를 활용한 저 방식은 현재 상태에서 seq의 최대값을 찾아낸 후, 그보다 1 높은 값을 추가하는 todo에 할당하는 방식입니다.